### ISSUE TITLE
argOder failed to handle function calls inside known call expressions | false positive

### ISSUE THREAD

**Issue by ssgreg** [go-critic/go-critic#963]
argOder failed to handle function calls inside known call expressions

```
cmd/agentd/main.go:174:25: argOrder: probably meant `strings.TrimSuffix(filepath.Ext(configFileName), configFileName)` (gocritic)
			config.SetConfigName(strings.TrimSuffix(configFileName, filepath.Ext(configFileName)))
```

there is no error in original call:
```
config.SetConfigName(strings.TrimSuffix(configFileName, filepath.Ext(configFileName)))
```

**Comment by cristaloleg** [comment]
Hi, sorry for the trouble, will look into the issue today.

**Comment by quasilyte** [comment]
It does look like an interesting case.

It's not a bug per se, it's just a pattern that happens to be valid. :)

argOrder is a quite simple checker. It warns if "usually non-const" argument is used in place of the "usually const" argument and vice versa. So, `strings.HasPrefix("http", url)` will be flagged.

The simple fix is to stop using const values info from types and handle only literal values. This way, your code will not be reported. It doesn't seem like we would lose a lot by sacrificing it anyway,.

**Comment by quasilyte** [comment]
This issue should be closed. Sorry for the delayed fix.

The next release will be out in a week or two as there are some other things I would like to fix before calling it a day. :)
